Sources/SwiftParser/Attributes.swift

```diff
@@ -176,7 +176,14 @@ extension Parser {
     argumentMode: AttributeArgumentMode,
     parseArguments: (inout Parser) -> RawAttributeSyntax.Arguments
   ) -> RawAttributeListSyntax.Element {
-    let (unexpectedBeforeAtSign, atSign) = self.expect(.atSign)
+    var (unexpectedBeforeAtSign, atSign) = self.expect(.atSign)
+    if atSign.trailingTriviaByteLength > 0 || self.currentToken.leadingTriviaByteLength > 0 {
+      let diagnostic = TokenDiagnostic(
+        self.swiftVersion < .v6 ? .extraneousTrailingWhitespaceWarning : .extraneousTrailingWhitespaceError,
```

Comment:
I don’t think that’s realistically possible. We need to have a different case in `SwiftSyntax.TokenDiagnostic` so that one of them can be marked as an error and the other one as a warning.